compiler/grainformat/reformat.re

```diff
@@ -110,95 +108,44 @@ let get_original_code =
   };
 };
 
-let last_file_loc = (): Grain_parsing.Location.t => {
-  let endpos: Stdlib__lexing.position = {
-    pos_fname: "",
-    pos_lnum: max_int,
-    pos_cnum: max_int,
-    pos_bol: 0,
-  };
-
-  let end_loc: Grain_parsing.Location.t = {
-    loc_start: endpos,
-    loc_end: endpos,
-    loc_ghost: true,
-  };
-  end_loc;
-};
-
-let make_line_start = (loc: Grain_parsing.Location.t) => {
-  let startpos: Stdlib__lexing.position = {
-    pos_fname: "",
-    pos_lnum: loc.loc_start.pos_lnum,
-    pos_cnum: 0,
-    pos_bol: 0,
-  };
-
-  let merged_loc: Grain_parsing.Location.t = {
-    loc_start: startpos,
-    loc_end: loc.loc_start,
-    loc_ghost: true,
-  };
-  merged_loc;
-};
-
-let make_to_line_end = (loc: Grain_parsing.Location.t) => {
-  let endpos: Stdlib__lexing.position = {
-    pos_fname: "",
-    pos_lnum: loc.loc_end.pos_lnum,
-    pos_cnum: max_int,
-    pos_bol: 0,
-  };
-
-  let merged_loc: Grain_parsing.Location.t = {
-    loc_start: loc.loc_end,
-    loc_end: endpos,
-    loc_ghost: true,
-  };
-  merged_loc;
-};
+let force_break_if_line_comment =
+    (comments: list(Parsetree.comment), separator) => {
+  switch (comments) {
+  | [] => separator
+  | _ =>
+    let last_comment = List.nth(comments, List.length(comments) - 1);
```

Comment:
A good question, and I wondered the same.    The number of comments in the list should be short, as this is usually scoped to a statement , expression etc.  Worth changing do you think?

Reply:
I want to defer to @ospencer because maybe he has more understanding of the implications of each.

Reply:
This is probably much faster since it avoids all the allocations of creating a new list if you do `List.rev`. But of course, the fastest would be a recursive function that just pulls the last item 🙂